.eslintrc

```diff
@@ -1,51 +1,51 @@
 {
   "root": true,
   "parser": "@typescript-eslint/parser",
-  "plugins": ["import", "prettier", "react-hooks", "@typescript-eslint"],
+  "plugins": ["import", "prettier", "react-hooks", "@typescript-eslint", "unused-imports"],
   "env": {
     "node": true
   },
   "extends": [
-    "prettier",
     "eslint:recommended",
     "plugin:@typescript-eslint/eslint-recommended",
-    "plugin:@typescript-eslint/recommended"
+    "plugin:@typescript-eslint/recommended",
+    "plugin:prettier/recommended"
   ],
   "rules": {
     "arrow-body-style": ["error", "as-needed"],
     "react/react-in-jsx-scope": "off",
-    "no-trailing-spaces": "error",
     "react/no-children-prop": "off",
     "no-console": ["warn", { "allow": ["warn", "error"] }],
-    "quotes": ["error", "single"],
     "import/order": [
       "error",
-        {
-          "groups": [
-            ["external", "builtin"],
-            "internal",
-            ["sibling", "parent"],
-            "index"
-          ],
-          "pathGroups": [
-            {
-              "pattern":
-                  "@(api|assets|modules|redux|routes|shared|resources)/**",
-              "group": "internal"
-            },
-            {
-              "pattern":
-                  "@(redux|routes|resources|api|i18n)",
-              "group": "internal"
-            }
-          ],
-          "pathGroupsExcludedImportTypes": ["internal"],
-          "newlines-between": "always"
+      {
+        "groups": [["external", "builtin"], "internal", ["sibling", "parent"], "index"],
+        "pathGroups": [
+          {
+            "pattern": "@(react)",
+            "group": "external",
+            "position": "before"
+          },
+          {
+            "pattern": "@(api|assets|modules|redux|routes|shared|resources)/**",
+            "group": "internal"
+          },
+          {
+            "pattern": "@(redux|routes|resources|api|i18n)",
+            "group": "internal"
+          }
+        ],
+        "pathGroupsExcludedImportTypes": ["internal", "react"],
+        "newlines-between": "always",
+        "alphabetize": {
+          "order": "asc",
+          "caseInsensitive": true
         }
+      }
     ],
     "import/no-cycle": "error",
     "react-hooks/rules-of-hooks": "error",
-    "react-hooks/exhaustive-deps": "off",
+    "react-hooks/exhaustive-deps": "warn",
```

Comment:
In numerous cases, not all arguments are necessary to be included in the dependencies of the useEffect hook. Implementing this rule might result in an abundance of unexpected warnings, which could be perceived as excessive.

Reply:
It will lead to add exceptional cases when we don't need to add dependencies as deps hooks because of existed business logic. For example, we need to add some useEffect on mount / unmount and to highlight it we don't need to add dependencies there.

Reply:
I'm not sure I understand the comment. The cases where it is necessary to exclude dependencies from the array should be the exception, and in those cases the rule can be disabled. Not having warnings at all increases the likelihood that we miss cases where omitting dependencies isn't the intention.

I don't see the high number of warning as a problem, because we won't block PRs for warnings. It is simply a signal to the developer (and reviewer) that you should pay attention here

Reply:
Missing dependencies can be incredibly annoying to debug, and I agree with @sultanofcardio that this rule is essential, even if enabled just as a warning for now (though if it were up to me, I'd want it to issue an error).

If you're certain you want to exclude a variable from the dependencies, rather than disable this rule entirely via a `// es-disable-…` comment, you can opt to do a more surgical approach by placing that variable in a ref, for example:

```tsx
export const MyComponent: React.FC<MyComponentProps> = ({ sideEffect }) => {
  const [count, setCount] = useState(0);
  const sideEffectRef = useRef(sideEffect); // 👈 Place unchanging variable in a ref
  
  const handleClick = useCallback(() => {
    setCount(count + 1);
    sideEffectRef.current();
  }, [count]); // 👈 Now we can preserve the exhaustive deps rule here
  
  return <h1 onClick={handleClick}>It Worked: {count)</h1>;
}；
```

If running an effect on mount/unmount is common, one way to reduce instances of `// es-disable-…` is to define a custom hook that is explicit in its purpose, for example:

```tsx
export const useOnceEffect = (callback: React.EffectCallback) => {
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(callback, []);
};
```